### PR TITLE
Implement disabling of caching private responses

### LINKED ISSUE #224: Support shared cache mode
It seems that hishel is currently caching requests with the header `Cache-Control: private` without a toggle to turn that off.

The use-case is some sort of caching proxy server for my API. The proxy takes the clients request and forwards them to a central server. That request from the proxy to the central server is cached on the proxy side using hishel. Having a way to prevent caching of user-specific data inside the proxy could be useful if I end up with some at one point. Otherwise, the only solution would be to set `no-store` for them.

### PR DESCRIPTION
**Pull request by Simon-Will**
I implemented the simple disabling of caching private responses that was suggested in #224 in order to support creating a shared cache.

I didn't implement any special handling for the case where a server specifies that only some headers are supposed to be private. Instead, any response with `private` in the `Cache-Control` is considered not cacheable.

Closes #224.

### CODE REVIEW

File: tests/test_controller.py [hunk 1]
```diff
@@ -160,6 +160,34 @@ def test_is_cachable_for_no_store():
...
+
+    assert controller.is_cachable(request=request, response=response)
+
+    response = Response(200, headers=[(b"Cache-Control", b"private")])
+
+    assert not controller.is_cachable(request=request, response=response)
+
+    response = Response(200, headers=[(b"Cache-Control", b"private=set-cookie")])
+
+    assert not controller.is_cachable(request=request, response=response)
```
Comment (Simon-Will) [inline tests/test_controller.py:178]: I was unsure if I should make different test cases for this (maybe even using `pytest.mark.parametrize`), but saw that there were some other tests using multiple asserts. So I did the same.

**Comment by Simon-Will** [comment]
Actually, I think this behavior also deserves a mention in the docs. Let me add that.

File: tests/test_controller.py (hunk 1 above)
Comment (karpetrosyan) [reply tests/test_controller.py:178]: It's okay, I like it

**Comment by karpetrosyan** [comment]
Looks very good, thanks! Can we also have a changelog entry for this feature?

**Comment by Simon-Will** [comment]
Certainly! I added a line to the changelog.

**Review by karpetrosyan — APPROVED** [review]

**Pull request merged**